src/program_executor.cpp

```diff
@@ -144,6 +144,17 @@ void ProgramExecutionServer::Execute(
       return;
     }
   }
+
+  moveit_msgs::CollisionObject surface;
```

Comment:
Putting the code here means that it won't get called if the program fails or is cancelled. Note that we call `Cancel` above for a variety of error conditions.

Could you please refactor the code as such:
- Add a `Finish()` method that removes the table from the planning scene and calls `PublishIsRunning(false);`
- Add a call to `Finish()` to the end of `Cancel()`
- Replace the call to `PublishIsRunning(false)` below (before `server_.setSucceeded()` with a call to `Finish()`.

The documentation for Finish will say (in program_executor.h):
```cpp
// Runs all necessary steps to finish up a program execution, regardless of whether it succeeded or failed.
```

Reply:
Requested changes are added. Please note that I did not add Finish() in the case of invalid program, because the invalid program does not run.